onnxruntime/core/providers/webnn/builders/impl/gru_op_builder.cc

```diff
@@ -219,6 +221,33 @@ bool GruOpBuilder::HasSupportedInputsImpl(const Node& node, const emscripten::va
   return IsDataTypeSupportedByOp(op_type, input0_type, wnn_limits, "input", "X", logger);
 }
 
+bool LstmOpBuilder::HasSupportedOutputsImpl(const Node& node,
+                                            const emscripten::val& wnn_limits,
+                                            const logging::Logger& logger) const {
+  const auto& output_defs = node.OutputDefs();
+  const auto& op_type = node.OpType();
+  int32_t Y_type = 0;
+  int32_t Y_h_type = 0;
+  bool has_Y = output_defs.size() > 0 && output_defs[0]->Exists();
+  bool has_Y_h = output_defs.size() > 1 && output_defs[1]->Exists();
+
+  if (has_Y && !has_Y_h && GetType(*output_defs[0], Y_type, logger)) {
+    return IsDataTypeSupportedByOp(op_type, Y_type, wnn_limits, "outputs", "Y", logger);
```

Comment:
It may return early if this returns `True`.

Reply:
Gru has two outputs (Y, Y_h). The if statement here enumerates all the cases. If the first if here returns `True`, this is expected. WDYT?

Reply:
OK, but this would introduce duplicates `GetType` for a single output.

How about just one by one check the outputs, and finally check if has both outputs, they should have same type.